client/src/test/java/io/pravega/client/stream/impl/ReaderGroupStateManagerTest.java

```diff
@@ -63,6 +63,56 @@ public MockControllerWithSuccessors(String endpoint, int port, ConnectionFactory
         }
     }
     
+    @Test(timeout = 10000)
+    public void testCompaction() throws ReinitializationRequiredException {
```

Comment:
Consider adding some commentary or structure to the test code, to relate it to the test goals:
1. validate that the state is identical before and after compaction
2. validate that the compaction timer is working as expected

I suppose the following is covered by core synchronizer tests:
1. validate that the revisioned stream mark is set and is effective at reducing the # of catch-up reads.